### ISSUE TITLE
test flake: omicron-nexus::test_all integration_tests::metrics::test_instance_watcher_metrics

### ISSUE THREAD

**Issue by sunshowers** [oxidecomputer/omicron#5752]
Example:

https://buildomat.eng.oxide.computer/wg/0/details/01HXT52WBZWPKP4HRFE5BZEJ90/TzkBQ5mcdy8cXJtRPqbP8sTUggD48AxTs1uwGGPeMg15q7e7/01HXT53AB77VSYQFJ33G6K5FSE

I feel like I saw something similar go by at some point.

**Comment by sunshowers** [comment]
Ahh looks like it was https://github.com/oxidecomputer/omicron/issues/5645.

cc @bnaecker maybe you have thoughts?

**Comment by iliana** [comment]
another: https://github.com/oxidecomputer/omicron/pull/5765/checks?check_run_id=24966472198

**Comment by bnaecker** [comment]
It looks like these tests are asking an instance to transition through various states; activating the instance-check background task; and then asserting the state of the metrics in ClickHouse. There's a race here between the collection task and the BG task publishing the metrics. While the code appears to wait until the BG task itself finishes, there's no guarantee that oximeter scrapes and inserts those metrics by the time the check is done.

Usually the trick here is to force oximeter to collect any outstanding metrics. The test context has access to the oximeter collector instance, and the `Oximeter::force_collect()` method will ensure that it polls all its producers. It will not return until those collections complete, either because they failed or because the data is in ClickHouse. There are several examples of that in the same test file where these failing tests are.

cc @hawkw

**Comment by hawkw** [comment]
> Usually the trick here is to force oximeter to collect any outstanding metrics. The test context has access to the oximeter collector instance, and the `Oximeter::force_collect()` method will ensure that it polls all its producers. It will not return until those collections complete, either because they failed or because the data is in ClickHouse. There are several examples of that in the same test file where these failing tests are.
> 
> cc @hawkw

Ah, thanks, I think I missed that. I'll go and fix it shortly.

**Comment by hawkw** [comment]
#5768 should fix this but it's waiting for #5767 to debreak the build first :/

**Comment by iliana** [comment]
well, hm. https://github.com/oxidecomputer/omicron/runs/24977844070

https://buildomat.eng.oxide.computer/wg/0/details/01HXWWGSG3442MM3JRMCY29Q72/ylQBEwmodRd8cuE6MefDsEoP7HdpgJcoPUrQjKYzxhrgvXIt/01HXWWH2AW89XR7MRJMGVFXCY6#S4469

```
thread 'integration_tests::metrics::test_instance_watcher_metrics' panicked at nexus/tests/integration_tests/metrics.rs:563:5:
assertion `left == right` failed
  left: 3
 right: 2
```
